scripts/pylib/twister/twisterlib.py

```diff
@@ -468,6 +471,40 @@ def _set_skip_reason(self, harness_state):
                 harness_testcase_result_skip in self.instance.results.values():
             self.instance.reason = "ztest skip"
 
+    def _verify_ztest_suite_name(self, harness_state, detected_suite_names, handler_time):
+        """
+        When test is written in ztest framework it should be also verify if
+        printed suite names correspond to expected suite names found in test
+        source file. Due to the fact, that Twister parse only c file and there
+        could be several test suites depend from #ifdef macro some of them could
+        be intentionally skipped. Due to that fact in this function there is
+        only check if detected suite names correspond to expected suite names
+        (and not in reverse).
+        """
+        expected_suite_names = self.instance.testcase.ztest_suite_names
+        harness_state_pass = "passed"
+        if not expected_suite_names or \
+                not harness_state == harness_state_pass:
+            return
+        if not detected_suite_names:
+            self._missing_suite_name(expected_suite_names, handler_time)
+        for detected_suite_name in detected_suite_names:
```

Comment:
As above. Can there be multiple detected suite names?

Reply:
Yes, there can be multiple detected suite names.